### PR TITLE
feat: improve seed words split

### PR DESCRIPTION
**Pull request by pedroferreira1**

### CODE REVIEW

File: src/wallet.js [hunk 1]
```diff
@@ -67,24 +67,28 @@ const wallet = {
    *
    * @param {string} words Words (separated by space) to generate the HD Wallet seed
    *
-   * @return {Object} {'valid': boolean, 'message': string}
+   * @return {Object} {'valid': boolean, 'message': string, 'words': cleanWords}
```
Comment (obiyankenobi) [inline src/wallet.js:70]: `words` is type `string`, right?

Also, it might be more useful to just return an array, no?

**Review by obiyankenobi — APPROVED** [review]

File: src/wallet.js [hunk 2]
```diff
@@ -67,24 +67,28 @@ const wallet = {
...
    * @memberof Wallet
    * @inner
    */
   wordsValid(words) {
+    let newWordsString = '';
     if (_.isString(words)) {
-      if (words.split(' ').length !== 24) {
+      // 1. Remove one or more spaces (or line breaks) before and after the 24 words
+      // 2. Substitute more then one space (or line break) for a single space
+      newWordsString = words.trim(/\s+/).replace(/\s+/g, ' ');
```
Comment (msbrogli) [inline src/wallet.js:79]: Maybe we should include `\r`, `\n`, and `\t`.

**Review by msbrogli — CHANGES_REQUESTED** [review]

File: src/wallet.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (pedroferreira1) [reply src/wallet.js:79]: Would be something like this including this comment and the other below?

`words.trim(/(\s\t\n)+/g).replace(/[^\x00-\x7F]/g, " ").replace(/\s+/g, ' ');`?

File: src/wallet.js (hunk 1 above)
Comment (pedroferreira1) [reply src/wallet.js]: Yeah, I improved this comment. Thanks

File: src/wallet.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (msbrogli) [reply src/wallet.js:81]: Maybe this is simpler: `words.replace(/[^A-Za-z0-9]/g, ' ').replace(/\s+/g, ' ').trim()`. What do you think?

**Review by msbrogli — APPROVED** [review]

File: src/wallet.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (pedroferreira1) [reply src/wallet.js]: Yeah, that also works and it's more readable. Changed to that

Comment (pedroferreira1) [reply src/wallet.js]: Resolved in the other comments

**Pull request merged**